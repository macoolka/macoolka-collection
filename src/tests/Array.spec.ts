import A from '../Array';
import test from '../testSuite'
import { pipe } from 'fp-ts/lib/pipeable'
describe('Array', () => {
    test(A, { immutable: false })
    pipe(
        [1, 2],
        A.map(a => a)
    )
    it('reverse', () => {
        pipe(
            A.reverse(A.empty()),
            expect(A.empty()).toEqual
        )
        pipe(
            A.reverse([1,2,3]),
            expect(A.from([3, 2, 1])).toEqual
        )
    })
})
