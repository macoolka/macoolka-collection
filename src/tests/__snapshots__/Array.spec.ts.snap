// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Array Collection Reader deleteWith 1`] = `
Array [
  Object {
    "age": 23,
    "city": "newyork",
    "id": "3",
    "isFemale": true,
    "name": "taylor",
  },
  Object {
    "age": 53,
    "city": "london",
    "id": "4",
    "isFemale": true,
    "name": "sam",
  },
]
`;

exports[`Array Collection Reader getShow 1`] = `"[{ id: \\"1\\", name: \\"john\\" }, { id: \\"2\\", name: \\"zhang\\" }, { id: \\"3\\", name: \\"taylor\\" }, { id: \\"4\\", name: \\"sam\\" }]"`;

exports[`Array Collection Reader groupByOption 1`] = `
Object {
  "\\"beijing\\"": Array [
    Object {
      "age": 21,
      "city": "beijing",
      "id": "1",
      "isFemale": false,
      "name": "john",
    },
    Object {
      "age": 34,
      "city": "beijing",
      "id": "2",
      "isFemale": false,
      "name": "zhang",
    },
  ],
  "\\"london\\"": Array [
    Object {
      "age": 53,
      "city": "london",
      "id": "4",
      "isFemale": true,
      "name": "sam",
    },
  ],
  "\\"newyork\\"": Array [
    Object {
      "age": 23,
      "city": "newyork",
      "id": "3",
      "isFemale": true,
      "name": "taylor",
    },
  ],
}
`;

exports[`Array Collection Reader updateWith 1`] = `
Array [
  Object {
    "age": 21,
    "city": "beijing",
    "id": "1",
    "isFemale": false,
    "name": "test",
  },
  Object {
    "age": 34,
    "city": "beijing",
    "id": "2",
    "isFemale": false,
    "name": "test",
  },
  Object {
    "age": 23,
    "city": "newyork",
    "id": "3",
    "isFemale": true,
    "name": "taylor",
  },
  Object {
    "age": 53,
    "city": "london",
    "id": "4",
    "isFemale": true,
    "name": "sam",
  },
]
`;

exports[`List Collection Reader deleteWith 1`] = `
Array [
  Object {
    "age": 23,
    "city": "newyork",
    "id": "3",
    "isFemale": true,
    "name": "taylor",
  },
  Object {
    "age": 53,
    "city": "london",
    "id": "4",
    "isFemale": true,
    "name": "sam",
  },
]
`;

exports[`List Collection Reader getShow 1`] = `"[{ id: \\"1\\", name: \\"john\\" }, { id: \\"2\\", name: \\"zhang\\" }, { id: \\"3\\", name: \\"taylor\\" }, { id: \\"4\\", name: \\"sam\\" }]"`;

exports[`List Collection Reader groupByOption 1`] = `
Object {
  "\\"beijing\\"": Array [
    Object {
      "age": 21,
      "city": "beijing",
      "id": "1",
      "isFemale": false,
      "name": "john",
    },
    Object {
      "age": 34,
      "city": "beijing",
      "id": "2",
      "isFemale": false,
      "name": "zhang",
    },
  ],
  "\\"london\\"": Array [
    Object {
      "age": 53,
      "city": "london",
      "id": "4",
      "isFemale": true,
      "name": "sam",
    },
  ],
  "\\"newyork\\"": Array [
    Object {
      "age": 23,
      "city": "newyork",
      "id": "3",
      "isFemale": true,
      "name": "taylor",
    },
  ],
}
`;

exports[`List Collection Reader updateWith 1`] = `
Array [
  Object {
    "age": 21,
    "city": "beijing",
    "id": "1",
    "isFemale": false,
    "name": "test",
  },
  Object {
    "age": 34,
    "city": "beijing",
    "id": "2",
    "isFemale": false,
    "name": "test",
  },
  Object {
    "age": 23,
    "city": "newyork",
    "id": "3",
    "isFemale": true,
    "name": "taylor",
  },
  Object {
    "age": 53,
    "city": "london",
    "id": "4",
    "isFemale": true,
    "name": "sam",
  },
]
`;
